### PR TITLE
[mod_feed] Add date to layout & options

### PR DESCRIPTION
**Pull request by brianteeman**
PR for #14300

Adds two new options to the module to optionally display the feed and item dates

### Test instructions
Create a new module to display a feed
To test you can use this feed http://www.feedforall.com/blog-feed.xml

Check to ensure that the TWO date options are set to NO by default and publish the module

Check the module on the front end and you will see a regular field as before this PR

Now enable the Date options and you will see a date for the feed and a date for each item

Also check the admin feed module where the same changes have been made

### CODE REVIEW

File: language/en-GB/en-GB.mod_feed.ini [hunk 1]
```diff
@@ -7,10 +7,14 @@ MOD_FEED="Feed Display"
 MOD_FEED_ERR_CACHE="Please make cache folder writeable."
 MOD_FEED_ERR_FEED_NOT_RETRIEVED="Feed not found."
 MOD_FEED_ERR_NO_URL="No feed URL specified."
+MOD_FEED_FIELD_DATE_DESC="Show the publication date of the feed."
+MOD_FEED_FIELD_DATE_LABEL="Feed Date"
 MOD_FEED_FIELD_DESCRIPTION_DESC="Show the description text for the entire feed."
 MOD_FEED_FIELD_DESCRIPTION_LABEL="Feed Description"
 MOD_FEED_FIELD_IMAGE_DESC="Show the image associated with the entire feed."
 MOD_FEED_FIELD_IMAGE_LABEL="Feed Image"
+MOD_FEED_FIELD_ITEMDATE_LABEL="Publication Date"
```
Comment (Quy) [inline language/en-GB/en-GB.mod_feed.ini]: Swap lines 16 & 17.

File: modules/mod_feed/mod_feed.xml [hunk 2]
```diff
@@ -70,6 +70,18 @@
 				</field>
 
 				<field
+					name="rssdate"
+					type="radio"
+					label="MOD_FEED_FIELD_DATE_LABEL"
+					description="MOD_FEED_DATE_DESCRIPTION_DESC"
```
Comment (Quy) [inline modules/mod_feed/mod_feed.xml]: Change to `MOD_FEED_FIELD_DATE_DESC`

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 7a685a8ad42db032dc6f653dd1e504732dc7af9f<br /><br />Works as expected in Front- and Backend.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18213">issues.joomla.org/tracker/joomla-cms/18213</a>.</sub>

File: administrator/modules/mod_feed/mod_feed.xml [hunk 3]
```diff
@@ -58,6 +58,18 @@
 				</field>
 
 				<field
+					name="rssdate"
+					type="radio"
+					label="MOD_FEED_FIELD_DATE_LABEL"
+					description="MOD_FEED_DATE_DESC"
```
Comment (Quy) [inline administrator/modules/mod_feed/mod_feed.xml]: Should be `MOD_FEED_FIELD_DATE_DESC`

File: modules/mod_feed/mod_feed.xml [hunk 4]
```diff
@@ -70,6 +70,18 @@
 				</field>
 
 				<field
+					name="rssdate"
+					type="radio"
+					label="MOD_FEED_FIELD_DATE_LABEL"
+					description="MOD_FEED_DATE_DESC"
```
Comment (Quy) [inline modules/mod_feed/mod_feed.xml]: Should be `MOD_FEED_FIELD_DATE_DESC`

File: modules/mod_feed/tmpl/default.php [hunk 5]
```diff
@@ -70,6 +70,14 @@
 					</h2>
 			<?php
 		}
+		// Feed date
+		if ($params->get('rssdate', 1)) :
+			{; ?>
```
Comment (Quy) [inline modules/mod_feed/tmpl/default.php]: Use `:` or `{`, but not necessary to use both. Same for the closing `}` or `endif`

File: administrator/modules/mod_feed/tmpl/default.php [hunk 6]
```diff
@@ -56,13 +56,21 @@
...
+		// Feed title
 		if (!is_null($feed->title) && $params->get('rsstitle', 1)) : ?>
 			<h2 class="<?php echo $direction; ?>">
 				<a href="<?php echo str_replace('&', '&amp;', $rssurl); ?>" target="_blank">
 				<?php echo $feed->title; ?></a>
 			</h2>
-		<?php endif; ?>
+		<?php endif;
+		// Feed date
+		if ($params->get('rssdate', 1)) :
```
Comment (Quy) [inline administrator/modules/mod_feed/tmpl/default.php]: Use `:` or `{`, but not necessary to use both. Same for the closing `}` or `endif`

**Comment by ghost** [comment]
This is an easy Test. You don't need Developer-Knowledge for Test.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18213">issues.joomla.org/tracker/joomla-cms/18213</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 40316e1c2b4a07e2b8f8646c49c15ec695aa25b5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18213">issues.joomla.org/tracker/joomla-cms/18213</a>.</sub>

File: administrator/modules/mod_feed/tmpl/default.php [hunk 7]
```diff
@@ -56,12 +56,19 @@
...
+		// Feed title
 		if (!is_null($feed->title) && $params->get('rsstitle', 1)) : ?>
 			<h2 class="<?php echo $direction; ?>">
 				<a href="<?php echo str_replace('&', '&amp;', $rssurl); ?>" target="_blank">
 				<?php echo $feed->title; ?></a>
 			</h2>
+		<?php endif;
+		// Feed date
+		if ($params->get('rssdate', 1)) :
+			?>
```
Comment (Quy) [inline administrator/modules/mod_feed/tmpl/default.php]: Move `?>` to line 67 and dedent lines 69 to 71.

File: modules/mod_feed/tmpl/default.php [hunk 8]
```diff
@@ -70,6 +70,13 @@
 					</h2>
 			<?php
 		}
+		// Feed date
+		if ($params->get('rssdate', 1)) :
+			?>
```
Comment (Quy) [inline modules/mod_feed/tmpl/default.php]: Move `?>` to line 74.

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 67a56e271279667c76a469db0748a148f709bc38<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18213">issues.joomla.org/tracker/joomla-cms/18213</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 67a56e271279667c76a469db0748a148f709bc38<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18213">issues.joomla.org/tracker/joomla-cms/18213</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by brianteeman** [comment]
thanks

**Pull request merged**